pkg/endpointmanager/allocator.go

```diff
@@ -14,21 +15,27 @@ const (
 	maxID = idpool.ID(4095)
 )
 
-var (
-	pool = idpool.NewIDPool(minID, maxID)
-)
+type epIDAllocator struct {
+	pool *idpool.IDPool
+}
+
+func newEPIDAllocator() *epIDAllocator {
+	return &epIDAllocator{
+		pool: idpool.NewIDPool(minID, maxID),
+	}
+}
 
-// ReallocatePool starts over with a new pool. This function is only used for
+// reallocatePool starts over with a new pool. This function is only used for
 // tests and its implementation is not optimized for production.
-func ReallocatePool() {
+func (a *epIDAllocator) reallocatePool(t testing.TB) {
```

Comment:
`t` looks unused in the method. Importing `testing` in non `*_test.go` files is kind of unfortunate, so maybe we could avoid that?

Reply:
The idea was to ensure that we don't use this method outside of the test context; but it is used (via `RemoveAll`) from other package's tests, so putting it in `_test` wasn't an option. Is there some other way we can mark this as test-only code?

Reply:
Ah right. I didn't consider that this is transitively also used in other package's tests. In that case I don't think there's any other way than exposing the method in non-test code :disappointed: I know people seldomly read godoc comments and we want to prevent use of the method outside of test code, but having `testing` imports in non-test code is a bit of a code smell as well. At least having a comment on both `reallocatePool` and `RemoveAll` about the purpose of the `t` parameter would be nice.

Reply:
Ah right. I didn't consider that this is transitively also used in other package's tests. In that case I don't think there's any other way than exposing the method in non-test code :disappointed: I know people seldomly read godoc comments and we want to prevent use of the method outside of test code, but having `testing` imports in non-test code is a bit of a code smell as well. At least having a comment on both `reallocatePool` and `RemoveAll` about the purpose of the `t` parameter would be nice.

Reply:
Another option would be to `panic` on [`testing.Testing()`](https://pkg.go.dev/testing#Testing) being `true` in `RemoveAll`. That way we'd not have to pass down the `*testing.TB` (which someone could potentially still provide in production code :ninja:).

Reply:
I've figured out a solution which allows us to just remove `RemoveAll` from non-package local testing code, and hence I moved it to _test.go :)